src/module.ml

```diff
@@ -130,21 +130,108 @@ module Visibility = struct
   let is_private t = not (is_public t)
 end
 
+module Obj_dir = struct
```

Comment:
Would it be possible put this into its own file? These large modules are hard to read.